src/custom/state/orders/updater.ts

```diff
@@ -230,3 +230,61 @@ export function EventUpdater(): null {
 
   return null
 }
+
+const CHECK_EXPIRED_ORDERS_INTERVAL = 10000 // 10 sec
+
+export function ExpiredOrdersWatcher(): null {
+  const { chainId } = useActiveWeb3React()
+
+  const expireOrder = useExpireOrder()
+
+  const pendingOrders = usePendingOrders({ chainId })
+
+  // ref, so we don't rerun useEffect
+  const pendingOrdersRef = useRef(pendingOrders)
+  pendingOrdersRef.current = pendingOrders
+
+  // for displaying expired orders
+  const addPopup = useAddPopup()
+
+  useEffect(() => {
+    if (!chainId) return
+
+    const checkForExpiredOrders = () => {
+      // no more pending orders
+      // but don't clearInterval so we can restart when there are new orders
+      if (pendingOrdersRef.current.length === 0) return
+
+      const now = new Date()
+      const expiredOrders = pendingOrdersRef.current.filter(order => {
+        // validTo is either a Date or unix timestamp in seconds
+        const validTo = typeof order.validTo === 'number' ? new Date(order.validTo * 1000) : order.validTo
```

Comment:
Not from this PR, but, I think the need of having to reevaluate `new Date(order.validTo * 1000)` over and over is a consequence of saving the state in a serializable form. I really don't like to have our design be driven by a middleware that is convenient to use an automatically persist stuff.